tests/unit/test_common.cpp

```diff
@@ -359,3 +362,64 @@ TEST(CommonAPITest, VecSim_QueryResult_Iterator) {
     // Free the internal array pointer - the validation for success is by having no leaks.
     VecSimQueryResult_FreeArray(res_inner_array);
 }
+
+class SerializerTest : public ::testing::Test {
+protected:
+    ~SerializerTest() { remove(file_name.c_str()); }
+
+    std::streampos GetFileSize() {
+        std::ifstream file(file_name, std::ios::binary);
+        const auto begin = file.tellg();
+        file.seekg(0, std::ios::end);
+        const auto end = file.tellg();
+        file.close();
+
+        return end - begin;
+    }
+
+    std::string file_name;
+};
+TEST_F(SerializerTest, HNSWSerialzer) {
```

Comment:
```suggestion
TEST_F(SerializerTest, HNSWSerialzerErrors) {
```
Also, try to see if you can increase the coverage in some additional cases where we send invalid `params` to the new constructor (bad `M`, bad type etc...), and test the error message itself in addition to the exception type